### PR TITLE
Instructions to prevent CORS for Hot Module Replacement

### PR DESCRIPTION
**Pull request by lambertbeekhuis**

### CODE REVIEW

**Comment by weaverryan** [comment]
Hey! Can you open a new PR? It looks like something got messed up here :/.

**Comment by OskarStark** [comment]
@weaverryan I changed the base branch to 4.3, but we need to merge it in 4.2. But now the diff looks correct.

Can you please review the PR? Thank you 😃

**Comment by lambertbeekhuis** [comment]
@weaverryan: Yes, I originally made it to the 4.3 branch, but later moved it to 4.2. That gave the errors, sorry.

I saw later that in 4.0 and 4.1 there was documentation on the --disable-host-check option, but removed in 4.2. I took me quite some time to figure this out. You need the this --disable-host-check in case the dev-server and your domain/ip are different. Additionally, you need the --port option, but this might also be related to an issue, see https://github.com/symfony/webpack-encore/issues/565.

**Comment by javiereguiluz** [comment]
@Kocal could you please review this proposed change? Thanks!

**Comment by Kocal** [comment]
Yes of course, I'm working on it :)

File: frontend/encore/dev-server.rst [hunk 1]
```diff
@@ -39,5 +39,24 @@ option:
 HMR currently works for :doc:`Vue.js </frontend/encore/vuejs>`, but does *not* work
 for styles anywhere at this time.
 
+For Hot Module Replacement, CORS-errors can appear (Cross Origin Resource Sharing). To handle this, add the --disable-host-check and --port options to your command:
```
Comment (Kocal) [inline frontend/encore/dev-server.rst]:
```suggestion
For Hot Module Replacement, CORS-errors can appear (Cross Origin Resource Sharing). To handle this, add the ``--disable-host-check`` and ``--port`` options to your command:
```

What about this?

File: frontend/encore/dev-server.rst [hunk 2]
```diff
@@ -39,5 +39,24 @@ option:
...
+
+.. code-block:: terminal
+
+    $ yarn encore dev-server --port 8080 --disable-host-check --hot
+
+Or, alternatively, you can add the options to your package.json
```
Comment (Kocal) [inline frontend/encore/dev-server.rst]:
```suggestion
Or, alternatively, you can add the options to your package.json:
```

**Review by Kocal — COMMENTED** [review]
Maybe we can add a warning about `--disable-host-check` usage?

I remember we already did this for Webpack Encore usage in a virtual machine (https://github.com/symfony/symfony-docs/blob/master/frontend/encore/virtual-machine.rst#fix-invalid-host-header-issue)

**Comment by javiereguiluz** [comment]
@lambertbeekhuis thanks for your contribution!

@Kocal thanks for your review! I did the proposed changes while merging.

**Pull request merged**